src/mcts/search.cc

```diff
@@ -1096,9 +1093,10 @@ SearchWorker::NodeToProcess SearchWorker::PickNodeToExtend(
           continue;
         }
         // If root move filter exists, make sure move is in the list.
-        if (!root_move_filter_.empty() &&
-            std::find(root_move_filter_.begin(), root_move_filter_.end(),
-                      child.GetMove()) == root_move_filter_.end()) {
+        if (!search_->root_move_filter_.empty() &&
```

Comment:
Do we trust the optimizer to pull search_->root_move_filter_ outside of the loop to avoid the extra deref or should we do that manually?

Reply:
I couldn't make much sense from disassembly, but it seemed that there was an indirection inside the loop, so I moved it out.
But nowadays indirection itself is no longer considered a performance problem, usually it takes a fraction of a CPU cycle (it's indirection to cold memory that is a problem).